### DISCUSSION TITLE
Switching from Craco to Rsbuild

### DISCUSSION THREAD

**Discussion by hussain-nz** [web-infra-dev/rsbuild#2480] [Ideas]
Hello Rsbuild/Rspack team :).

We have successfully switched from Craco to Rsbuild. We have some amazing performance improvements:
- Migration was relatively simple and painless. Nearly everything was available (features, plugins, etc)
- Bundling time is around 8-17 times faster on CI CD
- HMR improved around 2-3 times
- Colleagues with 16gb ram can work with our project without issues (we don't need the crazy `--max-old-space-size` flags anymore for 8-16gb)
- Colleagues with slow Intel "U" (energy saving) processors have reasonable HMR

There are some caveats:
- I know that Rsbuild HMR is quite slow currently (despite it improving for us), it's around 4-6s. I understand that this should come down to 0-2s after this improvement from Rspack is ported into Rsbuild: https://github.com/web-infra-dev/rsbuild/issues/1590#issuecomment-1954238581
- There are some memory leaks when Rsbuild config or .env files are changed, I filed the issue here: https://github.com/web-infra-dev/rsbuild/issues/2288

Despite the caveat, the improvement is great.

Our project is mid sized with 800 files of source code and 54000 lines of real code. The internet doesn't seem to talk much about Rspack/Rsbuild and I haven't seen people do an actual migration and report on it. Only 1-2 reddit comments and there are no details of the improvement besides "it was good" or something vague.

I am really hoping that Rspack/Rsbuild become super popular with the community, thus my suggestion: I would like to have detailed report of the performance improvements so that others can use this to be inspired to migrate. Where is the best place to post this?

**Comment by chenjiahan** [comment]
Thank you for sharing! I agree with you that HMR performance and memory leaks are important issues that we need to address further.

For data on performance improvements, I think Twitter is a good platform to share it. In addition, you can also share this data with other users in Rspack's Discord 😄

**Reply by hussain-nz** [reply]
Ok will do :)
